i8_terminal/common/layout.py

```diff
@@ -23,7 +23,7 @@ def format_metrics_df(df: DataFrame, target: str) -> DataFrame:
             else metric.display_format,
             target,
         )(
-            locate("int" if metric.data_format == "unsigned_int" else metric.data_format)(locate("float")(metric.value) if metric.data_format == "int" or metric.data_format == "unsigned_int" else metric.value)  # type: ignore # noqa: E501
+            locate("int" if metric.data_format == "unsigned_int" else "str" if metric.data_format in ["categorical", "datetime", "bool"] else metric.data_format)(locate("float")(metric.value) if metric.data_format == "int" or metric.data_format == "unsigned_int" else metric.value)  # type: ignore # noqa: E501
```

Comment:
Why is `locate` used two time here? The logic is very complex and hard to follow. Please re-write the code. @sadeghhadjigi ask @mehrdadrafiei for help. I remember he has refactored a similar case in the past.